src/main/java/com/microsoft/azure/functions/worker/Application.java

```diff
@@ -122,6 +123,9 @@ public static void main(String[] args) {
         } else {
             try (JavaWorkerClient client = new JavaWorkerClient(app)) {
                 client.listen(app.getWorkerId(), app.getRequestId()).get();
+            } catch (GracefulTerminationException ex) {
```

Comment:
Hi Shreyas, I don't think the exception will be catch as it will already be caught here https://github.com/Azure/azure-functions-java-worker/blob/ac6cb815d9493a0fcef7dd87a3006da3623b8b18/src/main/java/com/microsoft/azure/functions/worker/handler/MessageHandler.java#L49 right.